### PR TITLE
Bullet stacks

### PR DESCRIPTION
**Pull request by thgvr**
## About The Pull Request
Partial port of https://github.com/Mojave-Sun/mojave-sun-13/pull/2076 and surrounding PRs
Adds bullet handfuls. You can only have the same type of ammunition in a handful. No mismatched stacks
Spent casings are not stackable.

Generally, the rule is:
* By default, stacks can contain 12 rounds.
* Pistol-equivalent stacks contain 15 rounds.
* Shotshell stacks contain 8 rounds.
* Rifle stacks contain 15 rounds.
* Large calibre stacks contain 5 rounds.

Also replaces old ammo boxes with storage boxes that come pre-filled with stacks of bullets like this:
image
Here's how the stacks look:
image
image

## Changelog

:cl:
add: Added bullet stacks, they allow you to stack ammo of the same type into a group.
balance: Changed old ammo boxes into storage items containing stacks of bullets
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
it's peam

File: _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm [hunk 1]
```diff
@@ -374,6 +374,7 @@
 	},
 /obj/item/gun/energy/lasercannon{
 	pixel_y = 5;
+
```
Comment (thgvr) [inline _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm]:
```suggestion

```
committing this doesn't... actually do anything, not sure what to do so im trying this

File: _maps/RandomRuins/LavaRuins/lavaland_surface_lava_canyon.dmm [hunk 2]
```diff
@@ -884,6 +884,7 @@
 "zb" = (
 /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner{
 	loot = list(/obj/item/melee/transforming/cleaving_saw,/obj/item/gun/energy/kinetic_accelerator,/obj/item/keycard/gatedrop/lavacanyon);
+
```
Comment (thgvr) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_lava_canyon.dmm]:
```suggestion

```

File: _maps/RandomRuins/RockRuins/rockplanet_budgetcuts.dmm [hunk 3]
```diff
@@ -2746,6 +2746,7 @@
 "WY" = (
 /obj/structure/frame/computer{
 	dir = 1;
+
```
Comment (thgvr) [inline _maps/RandomRuins/RockRuins/rockplanet_budgetcuts.dmm]:
```suggestion

```

**Comment by DrCrawler** [comment]
Clicking one stack with another should add the clicked stack to the one you clicked it with imo.

**Comment by Ratvarr** [comment]
1. Rubbershot box does not: ship out; or spawn with ammo inside
2. 5.56x42 CLIP ammo box crate says it will send 120 bullets; box it ships sends 4 stacks of 12 bullets. that's deadly little bullets

**Comment by thgvr** [comment]
yeah I haven't changed any amounts yet but it is adjustable per box

**Comment by Erikafox** [comment]
image
This is like. Genuinely awful. I think it would be better served as being something you take out of the box instead of the box being filled with them

**Comment by thgvr** [comment]
image
that's a bug

**Comment by Erikafox** [comment]
ouhh

**Comment by generalthrax** [comment]
You can get invisible 0 bullet stacks. Just activate 1 bullet bullet stacks in your hand.

**Comment by Sun-Soaked** [comment]
> image that's a bug

my cranial implant explodes
I should just do a regex find on all these

**Comment by thgvr** [comment]
This is now ready

File: code/modules/projectiles/boxes_magazines/ammo_loaders.dm [hunk 4]
```diff
@@ -0,0 +1,176 @@
+// .357 Speed Loaders
+
+/obj/item/ammo_box/a357
+	name = "speed loader (.357)"
+	desc = "A 7-round speed loader for quickly reloading .357 revolvers. These rounds do good damage with average performance against armor."
+	icon_state = "speedloader_357-6"
+	base_icon_state = "speedloader_357"
+	ammo_type = /obj/item/ammo_casing/a357
+	caliber = ".357"
+	max_ammo = 7
```
Comment (SomeguyManperson) [inline code/modules/projectiles/boxes_magazines/ammo_loaders.dm]:
```suggestion
	desc = "A 6-round speed loader for quickly reloading .357 revolvers. These rounds do good damage with average performance against armor."
	icon_state = "speedloader_357-6"
	base_icon_state = "speedloader_357"
	ammo_type = /obj/item/ammo_casing/a357
	caliber = ".357"
	max_ammo = 6
```

**Pull request merged**